backend/src/resources/todos/todo.router.ts

```diff
@@ -1,23 +1,46 @@
-import { NextFunction, Request, Response, Router } from 'express';
+import { Request, Response, Router } from 'express';
 import { Todo } from './todo.model';
 
-const todos: Todo[] = [];
+let todos: Todo[] = [];
 const todosRouter = Router();
 
-todosRouter.get('/', (req: Request, res: Response, next: NextFunction) => {
-  res.json({
-    message: 'Todo: Implement',
-  });
+todosRouter.get('/', (req: Request, res: Response) => {
+  res.json(todos);
 });
 
-todosRouter.post('/', (req: Request, res: Response, next: NextFunction) => {
-  const todo: Todo = {
-    id: Math.random().toString(36).substring(2, 7),
-    description: req.body.description,
-    isDone: false,
-  };
-  todos.push(todo);
-  res.json(todo);
+todosRouter.delete('/:id', (req: Request, res: Response) => {
+  if (todos.findIndex(todo => todo.id === req.params.id) !== -1) {
+    todos = todos.filter(todo => todo.id !== req.params.id);
+    res.json({ status: true });
+  } else {
+    res.json({ status: false });
+  }
+});
+
+todosRouter.patch('/:id', (req: Request, res: Response) => {
+  if (!req.body.isDone) {
```

Comment:
This check has an issue - it will also discard `false` as an invalid value, which is wrong. Improve this check such that:

1. It is ensured that `isDone` is sent in the body
2. `isDone` is either `true` or `false`

If either of the conditions is not met, the client must get an appropriate error message.